src/node/cose_common.h

```diff
@@ -67,56 +72,160 @@ namespace ccf::cose
       static_cast<const char*>(item.val.string.ptr) + item.val.string.len};
   }
 
-  static std::pair<std::string /* issuer */, std::string /* subject */>
-  extract_iss_sub_from_sig(const std::vector<uint8_t>& cose_sign1)
+  struct CwtClaims
   {
-    QCBORError qcbor_result;
-    QCBORDecodeContext ctx;
-    UsefulBufC buf{cose_sign1.data(), cose_sign1.size()};
-    QCBORDecode_Init(&ctx, buf, QCBOR_DECODE_MODE_NORMAL);
+    int64_t iat{};
+    std::string iss{};
+    std::string sub{};
+  };
 
-    QCBORDecode_EnterArray(&ctx, nullptr);
-    qcbor_result = QCBORDecode_GetError(&ctx);
-    if (qcbor_result != QCBOR_SUCCESS)
+  struct CcfClaims
+  {
+    std::string txid{};
+  };
+
+  struct CcfCoseReceiptPhdr
+  {
+    int alg{};
+    std::vector<uint8_t> kid{};
+    CwtClaims cwt{};
+    CcfClaims ccf{};
+    int vds{};
+  };
+
+  struct Leaf
+  {
+    std::vector<uint8_t> write_set_digest;
+    std::string commit_evidence;
+    std::vector<uint8_t> claims_digest;
+  };
+
+  struct MerkleProof
+  {
+    Leaf leaf;
+    std::vector<std::pair<int64_t, std::vector<uint8_t>>> path;
+  };
+
+  struct CcfCoseReceipt
+  {
+    CcfCoseReceiptPhdr phdr;
+    std::vector<uint8_t> merkle_root;
+  };
```

Comment:
This is already defined within namespace `ccf::cose`, so its full name is `ccf::cose::CcfCoseReceipt`.

Can we flatten the nmae, and nest `Phdr` inside?

```suggestion
  struct Receipt
  {
    struct Phdr
    {
      int alg{};
      std::vector<uint8_t> kid{};
      CwtClaims cwt{};
      CcfClaims ccf{};
      int vds{};
    } phdr;
    std::vector<uint8_t> merkle_root;
  };
```